packages/firestore/android/src/reactnative/java/io/invertase/firebase/firestore/ReactNativeFirebaseFirestoreModule.java

```diff
@@ -161,6 +164,28 @@ public void terminate(String appName, Promise promise) {
             });
   }
 
+  @ReactMethod
+  public void persistenceCacheIndexManager(String appName, int requestType, Promise promise) {
+    PersistentCacheIndexManager indexManager =
+        getFirestoreForApp(appName).getPersistentCacheIndexManager();
+    if (indexManager != null) {
+      switch (requestType) {
+        case 0:
+          indexManager.enableIndexAutoCreation();
+          break;
+        case 1:
+          indexManager.disableIndexAutoCreation();
+          break;
+        case 2:
+          indexManager.deleteAllIndexes();
+          break;
+      }
+    } else {
+      Log.d("RNFBFirestoreModule", "`PersistentCacheIndexManager` is not available.");
```

Comment:
Should it be a throw here?

Reply:
Yeah, I can update to a throw, it should never get to this point as we check if persistence has been enabled in the JS code 👍